src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -1184,6 +1184,37 @@ IEnumerable<int> M()
                 Diagnostic(ErrorCode.ERR_BadIteratorLocalType, "x").WithLocation(7, 26));
         }
 
+        [Fact]
+        public void RefReadonlyInSwitchCaseInIterator()
```

Comment:
>RefReadonlyInSwitchCaseInIterator [](start = 20, length = 33)

The LDM had discussed relaxing the requirements for embedded statements to permit a local declaration as an embedded statement. Please check that this fix also diagnoses a case like the following to ensure we don't miss it later

```csharp
    public IEnumerable<int> M1() {
        yield return 0;
        int y = 0;
        if (true)
            ref int x = ref y;
    }
```